include/ceed/jit-source/cuda/cuda-shared-basis-tensor-templates.h

```diff
@@ -0,0 +1,419 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and other CEED contributors.
+// All Rights Reserved. See the top-level LICENSE and NOTICE files for details.
+//
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This file is part of CEED:  http://github.com/ceed
+
+#include <ceed.h>
+
+typedef struct {
+  CeedInt t_id_x;
+  CeedInt t_id_y;
+  CeedInt t_id_z;
+  CeedInt t_id;
+  CeedScalar* slice;
+} BackendData;
+
+//------------------------------------------------------------------------------
+// 1D
+//------------------------------------------------------------------------------
+
+//------------------------------------------------------------------------------
+
+//------------------------------------------------------------------------------
+// E-vector -> single element
+//------------------------------------------------------------------------------
+template <int NUM_COMP, int P_1D>
```

Comment:
@nbeams, one change I had to make here that these aren't templated on the strides since `num_elem` isn't known at compile time for bases